lib/internal/streams/pipeline.js

```diff
@@ -57,7 +49,11 @@ function destroyer(stream, reading, writing, callback) {
     if (destroyed) return;
     destroyed = true;
 
-    destroyStream(stream, err);
+    // request.destroy just do .end - .abort is what we want
+    if (isRequest(stream)) return stream.abort();
+    if (isRequest(stream.req)) return stream.req.abort();
+    if (typeof stream.destroy === 'function') return stream.destroy(err);
+    if (typeof stream.close === 'function') return stream.close();
```

Comment:
Can we re-use the implementation from above here?

Reply:
How about now?